File: test/models/hedra_audio_start_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:my_web_app/models/hedra_audio_start.dart';

void main() {
  test('parses the supported silence and crop offsets', () {
    expect(parseHedraAudioStartMs('-1000'), -1000);
    expect(parseHedraAudioStartMs('2000'), 2000);
    expect(parseHedraAudioStartMs('-30000'), hedraAudioStartMinMs);
    expect(parseHedraAudioStartMs('30000'), hedraAudioStartMaxMs);
  });

  test('rejects invalid or out-of-range values', () {
    expect(parseHedraAudioStartMs(''), isNull);
    expect(parseHedraAudioStartMs('1.5'), isNull);
    expect(parseHedraAudioStartMs('-30001'), isNull);
    expect(parseHedraAudioStartMs('30001'), isNull);
    expect(validateHedraAudioStartMs('not-a-number'), isNotNull);
  });
}

File: supabase/functions/viral-video-ad-generator/hedra_audio_start.test.ts
import {
  assertEquals,
  assertThrows,
} from "https://deno.land/std@0.224.0/assert/mod.ts";

import {
  HEDRA_AUDIO_START_MAX_MS,
  HEDRA_AUDIO_START_MIN_MS,
  parseHedraAudioStartMs,
  withHedraAudioStartMs,
} from "./hedra_audio_start.ts";

Deno.test("defaults the Hedra audio start offset to zero", () => {
  assertEquals(parseHedraAudioStartMs(undefined), 0);
  assertEquals(parseHedraAudioStartMs(null), 0);
});

Deno.test("accepts silence and crop offsets inside the app safety range", () => {
  assertEquals(parseHedraAudioStartMs(-1000), -1000);
  assertEquals(parseHedraAudioStartMs(2000), 2000);
  assertEquals(parseHedraAudioStartMs(HEDRA_AUDIO_START_MIN_MS), -30_000);
  assertEquals(parseHedraAudioStartMs(HEDRA_AUDIO_START_MAX_MS), 30_000);
});

Deno.test("rejects fractions, strings, and offsets outside the safety range", () => {
  assertThrows(() => parseHedraAudioStartMs(1.5), Error, "integer");
  assertThrows(() => parseHedraAudioStartMs("1000"), Error, "integer");
  assertThrows(() => parseHedraAudioStartMs(-30_001), Error, "between");
  assertThrows(() => parseHedraAudioStartMs(30_001), Error, "between");
});

Deno.test("adds the documented Hedra snake-case parameter", () => {
  assertEquals(withHedraAudioStartMs({ type: "video" }, -1000), {
    type: "video",
    audio_start_ms: -1000,
  });
});
